### DISCUSSION TITLE
ModelView access for request

### DISCUSSION THREAD

**Discussion by abdawwa1** [aminalaee/sqladmin#607] [Q&A]
Hello there , how can i get access for request.session in a class that inherits from ModelView class ?

Ex : 

```
class UserProfileAdmin(ModelView, model=User):
     list_query = select(User).filter(User.id == request.session.get("user"))
```

**Comment by aminalaee** [comment]
It depends which part you really need this, is it to filter out items in the `list_query` ?

**Reply by abdawwa1** [reply]
I need this part for displaying records depending on the current logged in user , let me correct my example here 

```
class UserProfileAdmin(ModelView, model=UserProfile):
     list_query = select(UserProfile).filter(UserProfile.user_id == request.session.get("user_id"))
```

So if the current logged in user_id = 1 then i need to list the profile data of the user_id = 1

**Reply by aminalaee** [reply]
Yeah that looks like a very reasonable use-case. I will take a look into that.

**Comment by JettScythe** [comment]
You can do this with     
```
def list_query(self, request: Request) -> Select:
        return select(UserProfile).filter(UserProfile.user_id == request.session.get("user_id"))
```
